File: ELearningFrontEnd/src/app/dtos/report/report-action.req.ts
export class ReportActionReq{

    public actionType: string;
    public entityType: string;
    public reason: string;
    public entityId: number;
    public ids: number[];

    constructor(data: any) {
        this.actionType = data.actionType;
        this.entityType = data.entityType;
        this.reason = data.reason;
        this.entityId = data.entityId;
        this.ids = data.ids;
    }
}

File: ELearningBackEnd/src/main/java/com/example/elearningbackend/report/ReportActionReq.java
package com.example.elearningbackend.report;

import com.example.elearningbackend.common.EntityNameEnum;
import com.example.elearningbackend.history.action.ActionType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
public class ReportActionReq {

    private ActionType actionType;
    private EntityNameEnum entityType;
    private String reason;
    private long entityId;
    private Set<Long> ids = new HashSet<>();

}
